### PR TITLE
Feature: LiteDB server and Files.Core removal

### PR DESCRIPTION
**Pull request by hez2010**
**Resolved / Related Issues**
- Moving database to the server project to ensure exact one database connection existing.
- Bump dependencies except FluentFTP
- Also generate `Microsoft.UI.Content.ContentExternalOutputLink` from winmd on-the-fly to avoid breaking changes in CsWinRT
- Remove Files.Core, this project is not being shared between any other project so I don't see any reason why it should be a separate project than Files.App

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?

**Screenshots (optional)**
Add screenshots here.

### CODE REVIEW

File: src/Files.App.Launcher/Files.App.Launcher.vcxproj [hunk 1]
```diff
@@ -321,21 +321,23 @@
...
+    <Import Project="..\..\packages\Microsoft.Windows.CppWinRT.2.0.240111.5\build\native\Microsoft.Windows.CppWinRT.targets" Condition="Exists('..\..\packages\Microsoft.Windows.CppWinRT.2.0.240111.5\build\native\Microsoft.Windows.CppWinRT.targets')" />
+    <Import Project="..\..\packages\Microsoft.Windows.ImplementationLibrary.1.0.240122.1\build\native\Microsoft.Windows.ImplementationLibrary.targets" Condition="Exists('..\..\packages\Microsoft.Windows.ImplementationLibrary.1.0.240122.1\build\native\Microsoft.Windows.ImplementationLibrary.targets')" />
   </ImportGroup>
   <Target Name="EnsureNuGetPackageBuildImports" BeforeTargets="PrepareForBuild">
     <PropertyGroup>
-      <ErrorText>This project references a NuGet package that is not on this computer. To download those packages, use Restore NuGet Packages. For more information, see http://go.microsoft.com/fwlink/?LinkID=322105. The missing file is {0}.</ErrorText>
+      <ErrorText>このプロジェクトは、このコンピューター上にない NuGet パッケージを参照しています。それらのパッケージをダウンロードするには、[NuGet パッケージの復元] を使用します。詳細については、http://go.microsoft.com/fwlink/?LinkID=322105 を参照してください。見つからないファイルは {0} です。</ErrorText>
```
Comment (0x5bfa) [inline src/Files.App.Launcher/Files.App.Launcher.vcxproj:337]: Is there a reason why this is changed to Japanese?

File: src/Files.App.Server/Data/Enums/GroupOption.cs [hunk 2]
```diff
@@ -1,75 +1,75 @@
...
+namespace Files.App.Server.Data.Enums
 {
-	public enum GroupOption : byte
+	public enum GroupOption
 	{
 		/// <summary>
 		/// No grouping.
 		/// </summary>
-		None,
+		None = 0,
```
Comment (0x5bfa) [inline src/Files.App.Server/Data/Enums/GroupOption.cs:11]: If it starts from 0, is this neeed?

File: src/Files.App.Server/Data/Enums/SortOption.cs [hunk 3]
```diff
@@ -1,70 +1,70 @@
...
+namespace Files.App.Server.Data.Enums
 {
-	public enum SortOption : byte
+	public enum SortOption
 	{
 		/// <summary>
 		/// Sort by name.
 		/// </summary>
-		Name,
+		Name = 0,
```
Comment (0x5bfa) [inline src/Files.App.Server/Data/Enums/SortOption.cs:11]: As above

**Review by 0x5bfa — COMMENTED** [review]
Looks good overall.

File: src/Files.App.Launcher/Files.App.Launcher.vcxproj (hunk 1 above)
Comment (hez2010) [reply src/Files.App.Launcher/Files.App.Launcher.vcxproj:337]: This is changed by VS automatically when I bump the dependency. Will revert it.

**Review by 0x5bfa — APPROVED** [review]
Overall, look good!

File: src/Files.App.Server/Data/Enums/SortOption.cs (hunk 3 above)
Comment (hez2010) [reply src/Files.App.Server/Data/Enums/SortOption.cs:11]: This is to satisfy CsWinRT.

File: src/Files.App.Server/Data/Enums/GroupOption.cs (hunk 2 above)
Comment (hez2010) [reply src/Files.App.Server/Data/Enums/GroupOption.cs:11]: This is to satisfy CsWinRT.

File: src/Files.App/Helpers/Layout/LayoutPreferencesDatabaseManager.cs [hunk 4]
```diff
@@ -1,200 +1,159 @@
...
-					tmp.LayoutPreferencesManager = preferencesItem;
-					col.Update(tmp);
-				}
-				else
-				{
-					// Remove file tag
-					col.Delete(tmp.Id);
-				}
-			}
+				NormalMaxLength = entry.NormalMaxLength,
```
[the remark this answers is not in the document]
Comment (hez2010) [reply src/Files.App/Helpers/Layout/LayoutPreferencesDatabaseManager.cs:81]: This is to convert `LayoutPreferencesItem` from Files.App to `LayoutPreferencesItem` from Files.App.Server. We cannot just simply move the `LayoutPreferencesItem` from Files.App to Files.App.Server as it contains a lot of services related to UI.
And yeah, changes will be saved to the DB.

**Comment by hez2010** [comment]
PTAL

**Review by 0x5bfa — APPROVED** [review]
Litmus code wise. I"m not sure for layout managers changes validation.

File: src/Files.App/nupkgs/Microsoft.UI.Content/Microsoft.UI.Content.cs
Comment (yaira2) [inline src/Files.App/nupkgs/Microsoft.UI.Content/Microsoft.UI.Content.cs:1]: I think this is needed for the shell previews to work

Comment (hez2010) [reply src/Files.App/nupkgs/Microsoft.UI.Content/Microsoft.UI.Content.cs:1]: This has been changed to generate from the winmd on-the-fly. So we no longer need to keep the source here.

Comment (yaira2) [reply src/Files.App/nupkgs/Microsoft.UI.Content/Microsoft.UI.Content.cs:1]: I think it was a modified version but either way it seems to be working properly

**Pull request merged**